composeApp/src/commonMain/kotlin/world/respect/app/viewmodel/lessons/list/LessonListViewModel.kt

```diff
@@ -0,0 +1,95 @@
+import androidx.navigation.toRoute
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import org.jetbrains.compose.resources.getString
+import respect.composeapp.generated.resources.Res
+import respect.composeapp.generated.resources.lesson_list
+import world.respect.app.app.LessonDetail
+import world.respect.app.app.LessonList
+import world.respect.app.appstate.AppBarSearchUiState
+import world.respect.app.datasource.fakeds.FakeOpdsDataSource
+import world.respect.app.viewmodel.RespectViewModel
+import world.respect.datasource.DataLoadParams
+import world.respect.datasource.DataLoadResult
+import world.respect.datasource.opds.model.OpdsFacet
+import world.respect.datasource.opds.model.OpdsPublication
+import world.respect.datasource.opds.model.ReadiumLink
+import world.respect.navigation.NavCommand
+
+data class LessonListUiState(
+    val publications: List<OpdsPublication> = emptyList(),
+    val lessonFilter: List<OpdsFacet> = emptyList(),
+    val selectedFilterTitle: String? = null,
+    val link: List<ReadiumLink> = emptyList(),
+
+    )
+
+class LessonListViewModel(
+    savedStateHandle: SavedStateHandle
+) : RespectViewModel(savedStateHandle) {
+
+    private val opdsDataSource: FakeOpdsDataSource = FakeOpdsDataSource()
+
+    private val _uiState = MutableStateFlow(LessonListUiState())
+
+    val uiState = _uiState.asStateFlow()
+
+    private val route: LessonList = savedStateHandle.toRoute()
+
+    init {
+
+        viewModelScope.launch {
+            _appUiState.update {
+                it.copy(
+                    title = getString(resource = Res.string.lesson_list),
+                    searchState = AppBarSearchUiState(
+                        visible = true
+                    )
+                )
+            }
+            opdsDataSource.loadOpdsFeed(
+                url = route.url,
+                params = DataLoadParams()
+            ).collect { result ->
+                when (result) {
+                    is DataLoadResult -> {
+                        _uiState.update {
+                            it.copy(
+                                publications = result.data?.publications ?: emptyList(),
+                                lessonFilter = result.data?.facets ?: emptyList(),
+                                link = result.data?.links ?: emptyList()
+
+                            )
+                        }
+                    }
+                    else -> {}
+                }
+            }
+        }
+    }
+
+    fun onClickFilter(title: String) {
+        _uiState.update { it.copy(selectedFilterTitle = title) }
+    }
+    fun onClickLesson(publication: OpdsPublication) {
+        val selfLink = uiState.value.link.find { it.rel?.equals("self") == true }?.href
+        val publicationSelfLink = publication.links.find { it.rel?.equals("self") == true }?.href
```

Comment:
The ```publicationSelfLink``` itself is correct.

There is no need for selfLink  - route.url is the URL for the publication feed.